### PR TITLE
Update to .NET 8

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

**Review by gaurovgiri — APPROVED** [review]
Looks good to me

**Review by Shadowghost — APPROVED** [review]

File: Jellyfin.Api/Controllers/EnvironmentController.cs [hunk 1]
```diff
@@ -169,7 +169,7 @@ public IEnumerable<FileSystemEntryInfo> GetDrives()
             // Check if unc share
             var index = path.LastIndexOf(UncSeparator);
 
-            if (index != -1 && path.IndexOf(UncSeparator, StringComparison.OrdinalIgnoreCase) == 0)
+            if (index != -1 && path[0] == UncSeparator)
```
Comment (cvium) [inline Jellyfin.Api/Controllers/EnvironmentController.cs:172]: This will crash if path is ""

Comment (Bond-009) [reply Jellyfin.Api/Controllers/EnvironmentController.cs:172]: How? We already check if it contains `UncSeparator` once, so it must have at least length 1, right?

File: MediaBrowser.Model/Dlna/StreamBuilder.cs [hunk 2]
```diff
@@ -866,16 +861,16 @@ private StreamInfo BuildVideoItem(MediaSourceInfo item, MediaOptions options)
...
-                playlistItem.SetOption(qualifier, "audiochannels", audioStream.Channels.ToString() ?? string.Empty);
+                playlistItem.SetOption(qualifier, "audiochannels", audioStream.Channels?.ToString(CultureInfo.InvariantCulture) ?? string.Empty);
 
                 if (!string.IsNullOrEmpty(audioStream.Profile))
                 {
                     playlistItem.SetOption(audioStream.Codec, "profile", audioStream.Profile.ToLowerInvariant());
                 }
 
-                if (audioStream.Level != 0)
+                if (audioStream.Level.HasValue)
```
Comment (cvium) [inline MediaBrowser.Model/Dlna/StreamBuilder.cs:871]: This changes the behaviour?

File: Jellyfin.Networking/HappyEyeballs/HttpClientExtension.cs [hunk 3]
```diff
@@ -65,7 +65,7 @@ public static async ValueTask<Stream> OnConnect(SocketsHttpConnectionContext con
             // See https://github.com/dotnet/corefx/pull/29792/files#r189415885 for more details.
             if (await Task.WhenAny(tryConnectAsyncIPv6, Task.Delay(200, cancelIPv6.Token)).ConfigureAwait(false) == tryConnectAsyncIPv6 && tryConnectAsyncIPv6.IsCompletedSuccessfully)
             {
-                cancelIPv6.Cancel();
+                await cancelIPv6.CancelAsync().ConfigureAwait(false);
```
Comment (barronpm) [inline Jellyfin.Networking/HappyEyeballs/HttpClientExtension.cs:68]: Why is this awaited but the next line calling `GetAwaiter().GetResult()`?

Comment (Bond-009) [reply Jellyfin.Networking/HappyEyeballs/HttpClientExtension.cs:68]: `tryConnectAsyncIPv6` Is already completed, so `GetAwaiter().GetResult()` just gets the result

File: Jellyfin.Api/Controllers/EnvironmentController.cs (hunk 1 above)
Comment (cvium) [reply Jellyfin.Api/Controllers/EnvironmentController.cs:172]: True

**Review by crobibero — APPROVED** [review]

**Comment by Gauvino** [comment]
Why not changing the net version to 8 in dockerfile ?

**Review by cvium — APPROVED** [review]

**Review by barronpm — APPROVED** [review]

**Pull request merged**